connector/connect/common/src/main/protobuf/spark/connect/expressions.proto

```diff
@@ -54,6 +54,8 @@ message Expression {
     google.protobuf.Any extension = 999;
   }
 
+   // (Optional) Keep the information of the origin for this expression such as stacktrace.
+  Origin origin = 18;
```

Comment:
We already have something similar in `RelationCommon` called `source_info`.

If we do this PR we should mark the ID 1 in RelationCommon as reserved and remove the value. Then we should add it back as `Origin` with a new ID.

Now, for expressions, I think we should probably create a similar `ExpressionCommon` message and put it there.